src/linkgraph/linkgraph_gui.cpp

```diff
@@ -728,11 +730,13 @@ void LinkGraphLegendWindow::OnClick(Point pt, int widget, int click_count)
  */
 void LinkGraphLegendWindow::OnInvalidateData(int data, bool gui_scope)
 {
+	if (this->num_cargo != _sorted_cargo_specs.size()) {
+		this->Close();
```

Comment:
Close? Not re-init or something? Or are we in some sort of failure state if we get here?

Reply:
It needs to be deleted and reopened, not just reinited, and I'm not sure on the semantics of opening the current window while closing it at the same time.

And the only way that this could happen if someone is changing NewGRF mid-game which for industries and cargo types is going to break anyway, so I don't think it's important enough to worth dealing with.

Reply:
Closing sounds perfectly fine to me, given it's due to the basically unsupported changing of NewGRFs.